src/tracer/ideal_age_example.F90

```diff
@@ -340,27 +374,122 @@ subroutine ideal_age_tracer_column_physics(h_old, h_new, ea, eb, fluxes, dt, G,
   year = US%s_to_T*time_type_to_real(CS%Time) * Isecs_per_year
 
   do m=1,CS%ntr
-    if (CS%sfc_growth_rate(m) == 0.0) then
-      sfc_val = CS%young_val(m)
+
+    if (CS%growth_rate(m) == 0.0) then
+      young_val = CS%young_val(m)
     else
-      sfc_val = CS%young_val(m) * &
-          exp((year-CS%tracer_start_year(m)) * CS%sfc_growth_rate(m))
+      young_val = CS%young_val(m) * &
+        exp((year-CS%tracer_start_year(m)) * CS%growth_rate(m))
     endif
-    do k=1,CS%nkml ; do j=js,je ; do i=is,ie
-      if (G%mask2dT(i,j) > 0.0) then
-        CS%tr(i,j,k,m) = sfc_val
-      else
-        CS%tr(i,j,k,m) = CS%land_val(m)
-      endif
-    enddo ; enddo ; enddo
-  enddo
-  do m=1,CS%ntr ; if (CS%tracer_ages(m) .and. &
-                      (year>=CS%tracer_start_year(m))) then
-!$OMP parallel do default(none) shared(is,ie,js,je,CS,nz,G,dt,Isecs_per_year,m)
-    do k=CS%nkml+1,nz ; do j=js,je ; do i=is,ie
-      CS%tr(i,j,k,m) = CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt*Isecs_per_year
-    enddo ; enddo ; enddo
-  endif ; enddo
+
+    if (m == CS%BL_residence_num) then
+
+      if (CS%use_real_BL_depth) then
+        do j=js,je ; do i=is,ie
+          nk = floor(BL_layers(i,j))
+
+          do k=1,nk
+            if (G%mask2dT(i,j) > 0.0) then
+              CS%tr(i,j,k,m) = CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt*Isecs_per_year
+            else
+              CS%tr(i,j,k,m) = CS%land_val(m)
+            endif
+          enddo
+
+          k = MIN(nk+1,nz)
+
+          if (G%mask2dT(i,j) > 0.0) then
+            layer_frac = BL_layers(i,j)-nk
+            layer_frac = 0.9
+            CS%tr(i,j,k,m) = layer_frac * (CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt &
+                             *Isecs_per_year) + (1.-layer_frac) * young_val
+          else
+            CS%tr(i,j,k,m) = CS%land_val(m)
+          endif
+
+
+          do k=nk+2,nz
+          if (G%mask2dT(i,j) > 0.0) then
```

Comment:
Please indent the if blocks at lines 412-416 and lines 431-434 by 2 points relative to the surrounding do loop blocks, to adhere to the MOM6 code standards at github.com/mom-ocean/MOM6/wiki/Code-style-guide (and as is already being done for other if-test blocks in this same subroutine).